### PR TITLE
[Merged by Bors] - chore(GroupCohomology/LowDegree): add `mem_(one|two)Coboundaries_iff`

### PR DESCRIPTION
**Pull request by yu-yama**
- Add `FunLike` instances and extensionality lemmas for `groupCohomology.(one|two)Cocycles`
- Add `groupCohomology.(one|two)Cocycles_coe_mk` and `groupCohomology.(one|two)Cocycles_toFun_eq_coe` simp lemmas as in other `FunLike` types
- Add `groupCohomology.mem_(one|two)Coboundaries_iff` lemmas to ease rewriting `f ∈ groupCohomology.(one|two)Coboundaries` into an existential statement where `f` is a cocycle.
- Remove some `.1` (`Subtype.val`) to use the `FunLike` instances on `groupCohomology.(one|two)Cocycles`

[Zulip thread](https://leanprover.zulipchat.com/#narrow/channel/217875-Is-there-code-for-X.3F/topic/Rewriting.20.60f.20.E2.88.88.20groupCohomology.2E.28one.7Ctwo.29Coboundaries.60)

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by 101damnations** [comment]
LGTM, thanks!

File: Mathlib/RepresentationTheory/GroupCohomology/LowDegree.lean [hunk 1]
```diff
@@ -223,6 +223,12 @@ def twoCocycles : Submodule k (G × G → A) := LinearMap.ker (dTwo A)
 
 variable {A}
 
+instance : FunLike (oneCocycles A) G A := ⟨(·.val), Subtype.val_injective⟩
```
Comment (erdOne) [inline Mathlib/RepresentationTheory/GroupCohomology/LowDegree.lean:226]: Can you update the code below (e.g. `f.1 (g * h)` in `oneCocycles_map_mul_of_isTrivial`) to use this instance?

**Comment by erdOne** [comment]
Also please move the content of the PR message above the --- so that they appear in the commit message when merged.

File: Mathlib/RepresentationTheory/GroupCohomology/LowDegree.lean (hunk 1 above)
Comment (yu-yama) [reply Mathlib/RepresentationTheory/GroupCohomology/LowDegree.lean:226]: Thank you for the review. They are updated in the commit 2d277c58c4cb6e88baf5beb34ec9f666047cfdbf.

**Comment by yu-yama** [comment]
> Also please move the content of the PR message above the --- so that they appear in the commit message when merged.

The message is now updated to list all changes above the hyphens.

**Review by YaelDillies — APPROVED** [review]
Thanks for the PR! Here are a few minor comments, and apart from that

maintainer merge

**Comment by riccardobrasca** [comment]
Thanks!

bors d+

**Comment by yu-yama** [comment]
Thank you for your reviews. The suggestions are applied in commit ddff1215862f3994c6849e934fc8782378abbffd.

bors r+

**Pull request closed without merging**